python/link_prediction/link_prediction.py

```diff
@@ -0,0 +1,167 @@
+    learning_rate: float -> learning rate for optimizer
+    split_ratio: float -> split ratio between training and validation set. There is not test dataset because it is assumed that user first needs to create new edges in dataset to test a model on them.
+    epochs_printing: int ->  how often do you want to print results from your model? Results will be from validation dataset. 
+    node_features: str -> property name where the node features are saved.
+    device_type: str ->  If model will be trained using CPU or cuda GPU. Possible values are cpu and cuda.
+    """
+    num_of_layers: int = 10
+    layer_type: str = "GAT"
+    optimizer: str = "SGD"
+    learning_rate: float = 0.01
+    num_epochs: int = 200
+    split_ratio: float = 0.775
+    epochs_printing: int = 10
+    device_type: str = "cuda" 
+    node_features: str = "features1"
+
+
+
+##############################
+# global parameters
+##############################
+
+link_prediction_parameters: LinkPredictionParameters = LinkPredictionParameters()
+##############################
+# All read procedures
+##############################
+
+@mgp.read_proc
+def set_model_parameters(ctx: mgp.ProcCtx, parameters: mgp.Map) -> mgp.Record(status=mgp.Number, message=str):
+    """
+    Saves parameters to the global parameters link_prediction_parameters. Specific parsing is needed because we want enable user to call it with a subset of parameters, no need to send them all. We will use some
+    kind of reflection to most easily update parameters. NOTE: We don't do any checks on values sent in here, code will fail later if some unsupported parameter is sent.
+    :param ctx: Reference to the context execution.
+    :param parameters:
+        num_of_layers: int -> number of layers
+        layer_type: str -> layer type
+        num_epochs: int -> number of epochs for model training
+        optimizer: str -> can be one of the following: ADAM, SGD, AdaGrad...
+        learning_rate: float -> learning rate for optimizer
+        split_ratio: float -> split ratio between training and validation set. There is not test dataset because it is assumed that user first needs to create new edges in dataset to test a model on them.
+        epochs_printing: int ->  how often do you want to print results from your model? Results will be from validation dataset. 
+        node_features: str -> property name where the node features are saved.
+        device_type: str ->  If model will be trained using CPU or cuda GPU. Possible values are cpu and cuda.
+    :return: 1 if all sent parameters are successfully saved, 0 otherwise.
+    """
+    global link_prediction_parameters
+    for key, value in parameters.items():
+        if hasattr(link_prediction_parameters, key) is False:
+            return mgp.Record(status=0, message="No attribute " + key + " in class LinkPredictionParameters")
+        try:
+            link_prediction_parameters.__setattr__(key, value)
```

Comment:
Use non-private `setattr` function: https://www.programiz.com/python-programming/methods/built-in/setattr